Automattic/wp-calypso
# Setup guided flow redirects

Related to 7365-gh-Automattic/dotcom-forge

## Proposed Changes

The guided flow has a few cases where we want to redirect the users to another flow. 

1. If the user answers `Migrating or importing an existing site` we redirect to `/setup/import-hosted-site` 
2. If the user answer `Create newsletter` we redirect to `/setup/newsletter`
3. If the user answers `Sell something` and `Get a website built quickly` we redirect to `/start/do-it-for-me-store`
4. If the user answers `Get a website built quickly` we redirect to `/start/do-it-for-me`

![image](https://github.com/Automattic/wp-calypso/assets/7000684/82c40986-a6c6-49a1-84e9-5a38190aaa56)

## Why are these changes being made?
- The `SegmentationSurvey` didn't offer a way to bypass the navigation to the next page. To avoid navigation issues after redirection I extended it to have `skipNextNavigation` which is called with the question key and answers. This will easily allow us to customize how the navigation happens and avoid issues with history.



*

## Testing Instructions



* Visit `/start/guided`
* Verify that selecting `Migrating or importing an existing site` will redirect to `/setup/import-hosted-site` 
* Verify that selecting `Create newsletter` will redirect to `/setup/newsletter`
* Verify that by selecting `Sell something` and `Get a website built quickly` we redirect to `/start/do-it-for-me-store`
* Verify that by selecting `Get a website built quickly` we redirect to `/start/do-it-for-me`

## Pre-merge Checklist
